external_test_api.pl

```diff
@@ -69,43 +80,43 @@ sub startup {
   );
 
   # Mocked OAuth2 endpoints
-  $r->post(
-    '/oauth/test/token' => sub {
-      my $c = shift;
-      $c->render(
-        json => {
-          access_token  => 'fake_access_token',
-          expires_in    => 3600,
-          refresh_token => 'fake_refresh_token',
-          scope         => 'openid profile email',
-          token_type    => 'bearer',
-        },
-        status => 200
-      );
-    }
-  );
-  $r->get(
-    '/oauth/test/authorize' => sub {
-      my $c   = shift;
-      my $url = Mojo::URL->new($c->param('redirect_uri'));
-      $url->query->append(code  => 'fake_return_code');
-      $url->query->append(state => $c->param('state'));
-      $c->render(text => $c->tag('a', href => $url, sub { 'Connect' }));
-    }
-  );
-  $r->get(
-    '/oauth/test/userinfo' => sub {
-      my $c = shift;
-      $c->render(
-        json => {
-          email          => 'oauth2-user@example.com',
-          name           => 'OAuth2 Test User',
-          email_verified => 1,
-        },
-        status => 200
-      );
-    }
-  );
+  # $r->post(
+  #   '/oauth/test/token' => sub {
+  #     my $c = shift;
+  #     $c->render(
+  #       json => {
+  #         access_token  => 'fake_access_token',
+  #         expires_in    => 3600,
+  #         refresh_token => 'fake_refresh_token',
+  #         scope         => 'openid profile email',
+  #         token_type    => 'bearer',
+  #       },
+  #       status => 200
+  #     );
+  #   }
+  # );
+  # $r->get(
+  #   '/oauth/test/authorize' => sub {
+  #     my $c   = shift;
+  #     my $url = Mojo::URL->new($c->param('redirect_uri'));
+  #     $url->query->append(code  => 'fake_return_code');
+  #     $url->query->append(state => $c->param('state'));
+  #     $c->render(text => $c->tag('a', href => $url, sub {'Connect'}));
+  #   }
+  # );
+  # $r->get(
+  #   '/oauth/test/userinfo' => sub {
+  #     my $c = shift;
+  #     $c->render(
+  #       json => {
+  #         email          => 'oauth2-user@example.com',
+  #         name           => 'OAuth2 Test User',
+  #         email_verified => 1,
+  #       },
+  #       status => 200
+  #     );
+  #   }
+  # );
```

Comment:
Is commenting this out intentional?

Reply:
Yeah. I am not sure how to make this work quite yet as we have builtin code in BMO that allows BMO to authenticate against an external OAuth2 provider such as Github, etc. And the test case for that code expects externalapi.test to mock the OAuth2 endpoints. But the Duo Security API also uses the same endpoint naming and so I made the Duo Security testing take priority over the older OAuth2 authentication tests which we not currently use in production for anything. The other alternative would be to add config parameters to make the endpoints configurable for both OAuth2 and Duo Security so that they can be changed just when running the tests. I can do that now or as a later fix.